pkg/cmd/cli/cmd/create/deploymentconfig.go

```diff
@@ -0,0 +1,141 @@
+package create
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/spf13/cobra"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/meta"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/runtime"
+
+	"github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+)
+
+const (
+	DeploymentConfigRecommendedName = "deploymentconfig"
+
+	deploymentConfigLong = `
+Create a deployment config that uses a given image.`
+
+	deploymentConfigExample = `  # Create an nginx deployment config named my-nginx
+  $ %[1]s my-nginx --image=nginx`
+)
+
+type CreateDeploymentConfigOptions struct {
+	DC     *deployapi.DeploymentConfig
+	Client client.DeploymentConfigsNamespacer
+
+	Mapper       meta.RESTMapper
+	OutputFormat string
+	Out          io.Writer
+	Printer      ObjectPrinter
+}
+
+// NewCmdCreateServiceAccount is a macro command to create a new service account
+func NewCmdCreateDeploymentConfig(name, fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	o := &CreateDeploymentConfigOptions{Out: out}
+
+	cmd := &cobra.Command{
+		Use:     name + " NAME --image=IMAGE -- [COMMAND] [args...]",
```

Comment:
I don't see any special handling of dashes. You may want to have a look at https://github.com/kubernetes/kubernetes/pull/13917